web/thesauruses/ruby/exception_handling.json

```diff
@@ -0,0 +1,315 @@
+      "base_data_type_exception_type",
+      "invalid_data_type_exception_type",
+      "variable_not_found_exception_type",
+      "object_not_found_exception_type",
+      "index_out_of_bounds_exception_type",
+      "key_name_not_found_exception_type",
+      "invalid_encoding_exception_type"
+    ],
+    "Mathematical Errors/Exceptions": [
+      "base_mathematical_exception_type",
+      "overflow_exception_type",
+      "floating_point_exception_type",
+      "divide_by_zero_exception_type",
+      "cannot_cast_types_exception_type"
+    ],
+    "Memory Errors/Exceptions": [
+      "base_memory_exception_type",
+      "cannot_allocate_memory_exception_type",
+      "out_of_memory_exception_type",
+      "stack_overflow_exception_type",
+      "cannot_access_pointer_exception_type"
+    ],
+    "File Input/Output Errors/Exceptions": [
+      "base_file_exception_type",
+      "end_of_file_exception_type",
+      "file_does_not_exist_exception_type",
+      "file_already_exists_exception_type",
+      "item_is_actually_directory_exception_type",
+      "item_is_not_a_directory_exception_type",
+      "cannot_access_file_exception_type",
+      "drive_disconnected_exception_type",
+      "file_no_longer_available_exception_type",
+      "file_timeout_exception"
+    ],
+    "Network Input/Output Errors/Exceptions": [
+      "base_network_exception_type",
+      "connection_refused_exception_type",
+      "connection_broken_exception_type",
+      "invalid_hostname_exception",
+      "cannot_establish_connection_exception_type",
+      "connection_timed_out_exception_type",
+      "bad_certificate_exception_type",
+      "expired_certificate_exception_type"
+    ],
+    "Operating System Errors/Exceptions": [
+      "child_process_could_not_be_created_exception_type",
+      "child_process_unexpected_ending_exception_type",
+      "process_not_found_exception_type"
+
+    ],
+    "Other Errors/Exception Types": [
+      "import_not_found_exception_type",
+      "keyboard_interrupt_exception_type",
+      "syntax_exception_type",
+      "uncaught_signal_exception_type"
+    ],
+    "Warnings": [
+      "base_warning_type",
+      "deprecation_warning_type"
+    ]
+  },
+  "exception_handling": {
+    "statement_to_import_exception_handling": {
+      "not-implemented": "true"
+    },
+    "are_exception_types_functions": {
+      "code": "false"
+    },
+    "are_exception_types_objects": {
+      "code": "true"
+    },
+    "are_exception_types_gotos": {
+      "code": "false"
+    },
+    "base_exception_type": {
+      "code": "Exception"
+    },
+    "define_new_exception_type": {
+      "code": "class CustomException < Exception"
+    },
+    "try_or_operator": {
+      "not-implemented": "true"
+    },
+    "statement_to_start_trying_code": {
+      "code": "begin\n  # try statements"
+    },
+    "statement_to_start_catching_exceptions": {
+      "code": "rescue => exception\n  # catch statements"
+    },
```

Comment:
This would be like a statement to start a block of catching exceptions. (Like if it was something like "rescue: " then had other things after it.)  So I think this is probably not-implemented.

Reply:
Thank you for checking
I've modified it!